src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ModelFactoryAnalyzer.cs

```diff
@@ -173,8 +174,80 @@ private static bool IsUserDefinedModelType(ITypeSymbol typeSymbol)
                 return false;
             }
 
-            // Only consider class types - all types in the assembly are candidates
-            return typeSymbol.TypeKind == TypeKind.Class;
+            // Only consider class and struct types, but not enums
+            if (typeSymbol.TypeKind != TypeKind.Class && typeSymbol.TypeKind != TypeKind.Struct)
+            {
+                return false;
+            }
+
+            // Filter out client types - they are not models
+            if (typeSymbol.Name.EndsWith(ClientSuffix))
+            {
+                return false;
+            }
+
+            // Filter out types that can be easily instantiated (have public constructors with all properties settable)  
+            if (CanBeConstructedUsingPublicApis(typeSymbol))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private static bool CanBeConstructedUsingPublicApis(ITypeSymbol typeSymbol)
+        {
+            if (!(typeSymbol is INamedTypeSymbol namedType))
+            {
+                return false;
+            }
+
+            // Check if there is at least one public constructor
+            var publicConstructors = namedType.Constructors.Where(c => c.DeclaredAccessibility == Accessibility.Public).ToList();
```

Comment:
If there is a public constructor that does not accept all non-public properties, that should still be flagged.   Example:

```csharp
public class ExampleModel
{
    public int Age { get; }
    public string Name { get; }
    
    public ExampleProperty(string name)
    {
        Name = name;
    }
}
```

In this example, there is no way to set the `Age` property, so a model factory is needed.